biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -495,50 +606,60 @@ subroutine terminate_cohorts( currentSite, currentPatch, level )
                      currentPatch%area &
                      * SF_val_CWD_frac(c) * (1.0_r8 -  EDPftvarcon_inst%allom_agb_frac(currentCohort%pft)) 
              enddo
-
+             
              currentPatch%leaf_litter(currentCohort%pft) = currentPatch%leaf_litter(currentCohort%pft) + currentCohort%n* &
-                  (currentCohort%bl)/currentPatch%area
+                  (leaf_c)/currentPatch%area
+
              currentPatch%root_litter(currentCohort%pft) = currentPatch%root_litter(currentCohort%pft) + currentCohort%n* &
-                  (currentCohort%br+currentCohort%bstore)/currentPatch%area 
+                  (fnrt_c+store_c)/currentPatch%area 
+
 
              ! keep track of the above fluxes at the site level as a CWD/litter input flux (in kg / site-m2 / yr)
              do c=1,ncwd
                 currentSite%CWD_AG_diagnostic_input_carbonflux(c)  = currentSite%CWD_AG_diagnostic_input_carbonflux(c) &
-                     + currentCohort%n*(currentCohort%bdead+currentCohort%bsw) * &
+                     + currentCohort%n*(struct_c + sapw_c) * &
                      SF_val_CWD_frac(c) * EDPftvarcon_inst%allom_agb_frac(currentCohort%pft) * hlm_days_per_year / AREA
                 currentSite%CWD_BG_diagnostic_input_carbonflux(c)  = currentSite%CWD_BG_diagnostic_input_carbonflux(c) &
-                     + currentCohort%n*(currentCohort%bdead+currentCohort%bsw) * &
+                     + currentCohort%n*(struct_c + sapw_c) * &
                      SF_val_CWD_frac(c) * (1.0_r8 -  EDPftvarcon_inst%allom_agb_frac(currentCohort%pft))  * hlm_days_per_year / AREA
              enddo
              
              currentSite%leaf_litter_diagnostic_input_carbonflux(currentCohort%pft) = &
                   currentSite%leaf_litter_diagnostic_input_carbonflux(currentCohort%pft) +  &
-                  currentCohort%n * (currentCohort%bl) * hlm_days_per_year  / AREA
+                  currentCohort%n * (leaf_c) * hlm_days_per_year  / AREA
              currentSite%root_litter_diagnostic_input_carbonflux(currentCohort%pft) = &
                   currentSite%root_litter_diagnostic_input_carbonflux(currentCohort%pft) + &
-                  currentCohort%n * (currentCohort%br+currentCohort%bstore) * hlm_days_per_year  / AREA
+                  currentCohort%n * (fnrt_c + store_c) * hlm_days_per_year  / AREA
 
           end if
           
+          ! Zero out the state pools
+          call SetState(currentCohort%prt,leaf_organ,carbon12_species,0.0_r8)
+          call SetState(currentCohort%prt,fnrt_organ,carbon12_species,0.0_r8)
+          call SetState(currentCohort%prt,sapw_organ,carbon12_species,0.0_r8)
+          call SetState(currentCohort%prt,struct_organ,carbon12_species,0.0_r8)
+          call SetState(currentCohort%prt,repro_organ,carbon12_species,0.0_r8)
+          call SetState(currentCohort%prt,store_organ,carbon12_species,0.0_r8)
```

Comment:
So here we we set these pools we are using carbon12 (specific isotopic version) and above we are just using "all_carbon_species"  Here we would allow for isotopic differentiation?

Reply:
Right, good question.  When we are using SetState(), we are initializing a specific variable, we can't set a set of them.  But in most other cases, when we are querying, we just want to know the total mass, so we use "all_carbon_species" to signal that we want all carbon isotopes.

If you have a scenario, where on the FATES side of things you want to ask PARTEH for the leaf C13 for instance, you could  call prt%GetState(leaf_organ,carbon13_species).  ... Assuming that you have that isotope enabled in the specify PARTEH hypothesis you have turned on.